scaleway/resource_mnq_credential.go

```diff
@@ -0,0 +1,268 @@
+package scaleway
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	mnq "github.com/scaleway/scaleway-sdk-go/api/mnq/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const (
+	defaultMNQCredentialTimeout = 5 * time.Minute
+)
+
+func resourceScalewayMNQCredential() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayMNQCredentialCreate,
+		ReadContext:   resourceScalewayMNQCredentialRead,
+		UpdateContext: resourceScalewayMNQCredentialUpdate,
+		DeleteContext: resourceScalewayMNQCredentialDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create:  schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Read:    schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Update:  schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Delete:  schema.DefaultTimeout(defaultMNQCredentialTimeout),
+			Default: schema.DefaultTimeout(defaultMNQCredentialTimeout),
```

Comment:
Only timeout that are used should be listed.
Adding a timeout allows the user to override it even tho it is useless

Reply:
Credentials are indeed a synchronous resource. Timeout are not very useful here.